src/MongoDB.Driver.Core/Core/Operations/RetryableWriteOperationExecutor.cs

```diff
@@ -135,6 +137,20 @@ private static bool AreRetryableWritesSupported(ConnectionDescription connection
                 connectionDescription.IsMasterResult.ServerType != ServerType.Standalone;
         }
 
+        private static bool ShouldRetryWrites<TResult>(IRetryableWriteOperation<TResult> operation, RetryableWriteContext context)
+        {
+            var writeConcern = operation.WriteConcern;
+            if (writeConcern != null && !writeConcern.IsAcknowledged)
```

Comment:
This is the fix. If the operation is unacknowledged it should not be retried.

Reply:
Consider minor and probably debatable suggestion:

            return
                context.RetryRequested &&
                AreRetryableWritesSupported(context.Channel.ConnectionDescription) &&
                !context.Binding.Session.IsInTransaction &&
                (writeConcern?.IsAcknowledged ?? true);

my point, that all these 4 conditions have the same business meaning in the context of this method, so I would try to place them together.

I will also be ok with the current code.

Reply:
I like the refactor in the latest commit.

Reply:
I like it too